stix_shifter_utils/stix_translation/src/json_to_stix/json_to_stix_translator.py

```diff
@@ -190,169 +82,225 @@ def gen_dict_extract(self, key, var):
-            group = False
-            unwrap = False
-
-            # unwrap array of stix values to separate stix objects
-            if 'unwrap' in ds_key_def:
-                unwrap = True
-
-            if ds_key_def.get('cybox', self.cybox_default):
-                object_name = ds_key_def.get('object')
-                if 'references' in ds_key_def:
-                    references = ds_key_def['references']
+                if transformer:
+                    try: 
+                        value = transformer.transform(value)
+                        if value is None:
+                            return None
+                    except Exception:
+                        return None
+                
+                if references:
                     if isinstance(references, list):
-                        stix_value = []
+                        return_value = []
                         for ref in references:
-                            if unwrap:
-                                pattern = re.compile("{}_[0-9]+".format(ref))
-                                for obj_name in object_map:
-                                    if pattern.match(obj_name):
-                                        val = object_map.get(obj_name)
-                                        stix_value.append(val)
-                            else:
-                                val = object_map.get(ref)
-                                if not self._valid_stix_value(self.properties, key_to_add, val):
-                                    continue
-                                stix_value.append(val)
-                        if not stix_value:
-                            continue
+                            if not isinstance(value, list):
+                                value = [value]
+                            for i, _ in enumerate(value):
+                                parent_key_ind = self._get_tag_ind(ref, object_tag_ref_map, create_on_absence=False, unwrap=i)
+                                if parent_key_ind:
+                                    return_value.append(parent_key_ind)
                     else:
-                        if unwrap:
-                            stix_value = []
-                            pattern = re.compile("{}_[0-9]+".format(references))
-                            for obj_name in object_map:
-                                if pattern.match(obj_name):
-                                    val = object_map.get(obj_name)
-                                    stix_value.append(val)
-                        else:
-                            stix_value = object_map.get(references)
-                            if not self._valid_stix_value(self.properties, key_to_add, stix_value):
-                                continue
+                        return_value = self._get_tag_ind(references, object_tag_ref_map, create_on_absence=False)
+                        if unwrap is not False and not isinstance(return_value, list):
```

Comment:
there used to be a comment about `unwrap` flag. I would keep the comment here or somewhere else whatever is suitable.